src/parsers/phrase.ts

```diff
@@ -1,6 +1,50 @@
 import { Node } from '../b-tree';
 import { v4 as uuid } from 'uuid';
 
-export function parse<N extends Node>(str: string): N | void {
- // Implemented in JS
+export interface Position {
+  from: number;
+  to: number;
+}
+
+export interface Token {
+  id: string;
+  type: 'token';
+  position: Position;
+  text: string;
+}
+
+export interface Phrase {
+  id: string;
+  type: 'phrase';
+  position: Position;
+  text: string;
+  tokens: Token[];
+}
+
+export interface QuotedPhrase {
+  id: string;
+  type: 'quoted_phrase';
+  position: Position;
+  text: string;
+}
+
+export interface Relation {
+  id: string;
+  text: string;
+  value: string;
+  position: Position;
+}
+
+export interface Predicate {
```

Comment:
@Istar-Eldritch shouldn't this type replace the one here: https://github.com/repositive/query-parser/blob/master/src/b-tree/predicate.ts#L13